### PR TITLE
Build API here as well; rename to libopentelemetry-cpp{,-headers}

### LINKED ISSUE #38: Build shared libs, add run-export, register in global pinning
Hey all

I just noticed that arrow comes with an ability to build against opentelemetry, so it would be nice to have this lib ready for that use-case.

AFAICT from the various feedstocks floating around, this is the main one for the CPP lib, but it:
* build [static](https://github.com/regro/libcfgraph/blob/master/artifacts/cpp-opentelemetry-sdk/conda-forge/linux-64/cpp-opentelemetry-sdk-1.8.1-h367e6c9_0.json#L417-L439) instead of shared libs
* doesn't have a run-export
* is not part of the global [pinning](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml)

Those things would be the minimum necessary for arrow to build against this feedstock.

### PR DESCRIPTION
**Pull request by h-vetinari**
As discussed in #38; the api-feedstock can follow separately

Closes #38

The diff here is small when viewed with white-space changes ignored (only indentation increased for the most part; see also per-commit).

### CODE REVIEW

**Comment by h-vetinari** [comment]
@lidavidm @xhochy 
Any further comments here?

**Review by lidavidm — APPROVED** [review]
Ah, sorry, I thought this wasn't ready since it was in draft.

LGTM

**Comment by h-vetinari** [comment]
> Ah, sorry, I thought this wasn't ready since it was in draft.

It was in draft because I wanted to hear people's opinions first. 😅

**Comment by h-vetinari** [comment]
> the api-feedstock can follow separately

Actually, since both  `-sdk` & `-api` (as currently named) are built from the same sources, why not unify everything in this feedstock? That would make life a whole lot easier in several respects...

**Comment by lidavidm** [comment]
Sounds good to me. (...to be honest, when I first made these feedstocks, I don't think I realized multi-output recipes were a thing. Packaging for Conda is not something I do much.)

**Comment by h-vetinari** [comment]
Cool :)

Do you care particularly about the git-history of the api-feedstock? It's possible to import it but a bit messy. (Here's a recent [example](https://github.com/conda-forge/boost-feedstock/pull/164/commits) where the reasons to maintain the history outweigh the hassle & messiness).

**Comment by lidavidm** [comment]
I don't particularly care, so if it's a hassle I'd rather not bother.

**Comment by h-vetinari** [comment]
OK, this is ready for another peek @lidavidm. Basically I'm planning to merge it as soon as https://github.com/conda-forge/feedstock-outputs/pull/46 is in.

**Review by lidavidm — APPROVED** [review]

File: recipe/build.sh [hunk 1]
```diff
@@ -1,47 +1,40 @@
...
-main() {
-    # Release tarballs do not contain the required Protobuf definitions.
-    cp -r $CONDA_PREFIX/share/opentelemetry/opentelemetry-proto/opentelemetry ./third_party/opentelemetry-proto/
-    # Stop CMake from trying to git clone the Protobuf definitions.
-    mkdir ./third_party/opentelemetry-proto/.git
-    
-    mkdir -p build-cpp
-    pushd build-cpp
+# Release tarballs do not contain the required Protobuf definitions.
+cp -r $CONDA_PREFIX/share/opentelemetry/opentelemetry-proto/opentelemetry ./third_party/opentelemetry-proto/
```
Comment (xhochy) [inline recipe/build.sh:5]: For clarity
```suggestion
cp -r $BUILD_PREFIX/share/opentelemetry/opentelemetry-proto/opentelemetry ./third_party/opentelemetry-proto/
```

**Review by xhochy — APPROVED** [review]
Tiny nit, otherwise: nice!

**Pull request merged**